tasks/main.yml

```diff
@@ -215,16 +264,48 @@
       name: consul
       state: started
       enabled: yes
+    when: ansible_os_family != "Windows"
+
+  - name: Install NSSM
+    win_chocolatey:
+      name: nssm
+      state: latest
+
+  - name: Start Consul on Windows
+    win_nssm:
+      name: consul
+      state: started
+      application: "{{ consul_binary }}"
+      app_parameters_free_form: "agent -config-file='{{ consul_config_path }}/config.json' -config-dir='{{ consul_configd_path }}'"
+      stdout_file: "{{ consul_log_path }}/consul-nssm-output.log"
+      stderr_file: "{{ consul_log_path }}/consul-nssm-error.log"
+    when: ansible_os_family == "Windows"
 
   - name: Check Consul HTTP API
     wait_for:
       delay: 15
       port: 8500
+    when: ansible_os_family != "Windows"
+
+  - name: Check Consul HTTP API
```

Comment:
There's no `win_wait_for` module yet. This is a hack to use `wait_for` locally and watch the port in the same way. It works on my machine perfectly, but I can see cases where this could potentially fail depending on network setup. It may need more work in the future.